eldritch/test_encounters.py

```diff
@@ -1566,5 +1567,527 @@ def testIsle7Fail(self):
+    choice.resolve(self.state, "Yes")
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=2)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.dollars, 0)
+
+  def testStore7(self):
+    self.state.common.append(items.Dynamite())
+    # Why does a young child have dynamite?
+    self.state.event_stack.append(encounters.Store7(self.char))
+    self.resolve_until_done()
+    self.assertEqual(len(self.char.possessions), 1)
+    self.assertEqual(self.char.possessions[0].name, "Dynamite")
+
+
+class GraveyardTest(EncounterTest):
+
+#  def testGraveyard1(self):
+#    raise NotImplementedError("A monster appears")
+
+  def testGraveyard2Pass(self):
+    self.state.event_stack.append(encounters.Graveyard2(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 2)
+    self.assertEqual(self.char.clues, 1)
+
+  def testGraveyard2PassInsane(self):
+    self.char.sanity = 1
+    self.state.event_stack.append(encounters.Graveyard2(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 1)
+    self.assertEqual(self.char.place.name, "Asylum")
+    self.assertEqual(self.char.clues, 1)
+
+  def testGraveyard2Fail(self):
+    self.state.event_stack.append(encounters.Graveyard2(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=4)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.place.name, "Rivertown")
+
+  def testGraveyard3Pass(self):
+    self.char.fight_will_slider = 2
+    self.state.unique.append(items.HolyWater())
+    self.state.event_stack.append(encounters.Graveyard3(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=5)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.clues, 1)
+    self.assertEqual(len(self.char.possessions), 1)
+    self.assertEqual(self.char.possessions[0].name, "Holy Water")
+
+  def testGraveyard3FailOne(self):
+    self.char.fight_will_slider = 2
+    self.char.stamina = 4
+    self.state.unique.append(items.HolyWater())
+    self.state.event_stack.append(encounters.Graveyard3(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=1)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 3)
+
+  def testGraveyard3FailTwo(self):
+    self.char.fight_will_slider = 2
+    self.char.stamina = 4
+    self.state.unique.append(items.HolyWater())
+    self.state.event_stack.append(encounters.Graveyard3(self.char))
+    with mock.patch.object(events.random, "randint", new=mock.MagicMock(return_value=2)):
+      self.resolve_until_done()
+    self.assertEqual(self.char.stamina, 2)
+
+  def testGraveyard3FailThree(self):
```

Comment:
I probably wouldn't have bothered with all of these, but no reason not to leave them in now that they're there.